specifications/locks_auxiliary_vars/README.md

```diff
@@ -0,0 +1,36 @@
+# Two-way refinement using auxiliary variables
+
+Refinement proves that all behaviors implementation are a valid behavior of
+an abstract specification. It is usually not done from the abstract 
+specification to the implementation.
+
+This example shows an example of this second, unusual, case using auxiliary 
+variables, as presented in **Prophecy made simple** by *Lamport and Merz*.
+
+First a refinement from Peterson's algorithm to an abstract lock is provided
+in `Peterson.tla`. Second a refinement from an abstract lock with auxiliary
+variables is provided in `LockHS.tla`
+
+## Lock.tla
+
+The specification of an abstract lock for two processes in PlusCal
+
+## Peterson.tla
+
+The specification of Peterson's algorithm in PlusCal, and the refinement proof
+towards `Lock!Spec`
+
+## LockHS.tla
+
+The extension of `Lock.tla` using a history variable `h_turn` and a stutter
+variable `s`, provided by the `Stuttering.tla` module. 
+A refinement proof towards `Peterson!Spec` is presented within.
```

Comment:
Again, it would make sense to use TLC for checking the different theorems proved in this module?